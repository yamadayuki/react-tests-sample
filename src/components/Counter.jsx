import React, { useState } from "react";

export function Counter() {
  const [count, setCount] = useState(0);

  return (
    <div data-testid="counter">
      <h2>Counter</h2>
      <p data-testid="text">count: {count}</p>
      <button onClick={() => setCount(count + 1)} data-testid="button">
        +
      </button>
    </div>
  );
}
