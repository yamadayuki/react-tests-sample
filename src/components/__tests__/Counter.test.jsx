import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";
import { Counter } from "../Counter";

// see https://github.com/gnapse/jest-dom#usage
import "jest-dom/extend-expect";

// see https://jestjs.io/docs/ja/tutorial-react

// automatically unmount and cleanup DOM after the test is finished
afterEach(cleanup);

it("is visible for the users", () => {
  const { getByTestId } = render(<Counter />);

  expect(getByTestId("counter")).toBeVisible();
});

it("changes the counter text when the button is clicked", () => {
  const { getByTestId } = render(<Counter />);

  expect(getByTestId("text")).toHaveTextContent("count: 0");

  fireEvent.click(getByTestId("button"));

  expect(getByTestId("text")).toHaveTextContent("count: 1");
});
