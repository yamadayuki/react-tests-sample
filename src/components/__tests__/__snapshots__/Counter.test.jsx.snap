// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashes 1`] = `
<div
  data-testid="counter"
>
  <h2>
    Counter
  </h2>
  <p
    data-testid="text"
  >
    count: 
    0
  </p>
  <button
    data-testid="button"
  >
    +
  </button>
</div>
`;
